PetApiTest/PetApiTest.cs

```diff
@@ -0,0 +1,200 @@
+            await client.PostAsync("petStore/addNewPet", requestBody);
+            //when
+            var response = await client.GetAsync("petStore/pettype/dog");
+
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            List<Pet> actualPets = JsonConvert.DeserializeObject<List<Pet>>(responseString);
+            Assert.Equal(new List<Pet>() { pet }, actualPets);
+        }
+
+        [Fact]
+        public async Task Should_Return_Pet_When_Get_Pet_By_Color()
+        {
+            TestServer server = new TestServer(new WebHostBuilder()
+                .UseStartup<Startup>());
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("petStore/clear");
+            Pet pet = new Pet("Moon", "cat", "yellow", 1000);
+            string request = JsonConvert.SerializeObject(pet);
+            StringContent requestBody = new StringContent(request, Encoding.UTF8, "application/json");
+            await client.PostAsync("petStore/addNewPet", requestBody);
+            //when
+            var response = await client.GetAsync("petStore/petcolor/yellow");
+
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            List<Pet> actualPets = JsonConvert.DeserializeObject<List<Pet>>(responseString);
+            Pet dog = new Pet("Moon", "cat", "yellow", 1000);
+            Assert.Equal(new List<Pet>() { dog }, actualPets);
+        }
+
+        [Fact]
+        public async Task Should_Return_Pets_When_Get_Pets_By_Price_Range()
+        {
+            TestServer server = new TestServer(new WebHostBuilder()
+                .UseStartup<Startup>());
+            HttpClient client = server.CreateClient();
+            await client.DeleteAsync("petStore/clear");
+
+            Pet cat = new Pet("Moon", "cat", "yellow", 1000);
+            string requestCat = JsonConvert.SerializeObject(cat);
+            StringContent requestBodyCat = new StringContent(requestCat, Encoding.UTF8, "application/json");
+            await client.PostAsync("petStore/addNewPet", requestBodyCat);
+
+            Pet dog = new Pet("Sun", "dog", "yellow", 5000);
+            string requestDog = JsonConvert.SerializeObject(dog);
+            StringContent requestBodyDog = new StringContent(requestDog, Encoding.UTF8, "application/json");
+            await client.PostAsync("petStore/addNewPet", requestBodyDog);
+            //when
+            var response = await client.GetAsync("petStore/petpricerange/1000-5000");
+
+            //then
+            response.EnsureSuccessStatusCode();
+            var responseString = await response.Content.ReadAsStringAsync();
+            List<Pet> actualPets = JsonConvert.DeserializeObject<List<Pet>>(responseString);
+            Assert.Equal(new List<Pet>() { cat, dog }, actualPets);
```

Comment:
When design a test case for get pets by price range, not getting all items will be better